app/models/campaign_page.rb

```diff
@@ -27,4 +30,9 @@ class CampaignPage < ActiveRecord::Base
   def create_slug
     self.slug = title.parameterize if slug.nil? and not title.nil?
   end
+
+  # Compiles the HTML for this CampaignPage so that it can be used by external display apps.
+  def compile_html
+    self.compiled_html = render partial: 'campaign_pages/page_compile', layout: false, locals: {campaign_page: self}
```

Comment:
Thanks, that was the feedback I was looking for. I'll make that change when
I get home.
On Jul 16, 2015 11:26 AM, "Omar Sahyoun" <notifications@github.com> wrote:

> In app/models/campaign_page.rb
> <https://github.com/SumOfUs/Champaign/pull/59#discussion_r34800074>:
>
> > @@ -27,4 +30,9 @@ class CampaignPage < ActiveRecord::Base
> >    def create_slug
> >      self.slug = title.parameterize if slug.nil? and not title.nil?
> >    end
> > +
> > +  # Compiles the HTML for this CampaignPage so that it can be used by external display apps.
> > +  def compile_html
> > +    self.compiled_html = render partial: 'campaign_pages/page_compile', layout: false, locals: {campaign_page: self}
>
> I'm not comfortable having a model compile its own HTML - it's a fat
> violation of SRP.
>
> Really there should be a dedicated class for this:
>
> class CampaignPageRender
>   include RenderAnywhere
>
>   def initialize(page)
>     @page = page
>   end
>
>   def render_and_save
>     @page.update_attributes(compiled_html: html)
>   end
>
>   private
>
>   def html
>     render partial: 'campaign_pages/page_compile', layout: false, locals: {campaign_page: @page}
>   endend
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SumOfUs/Champaign/pull/59/files#r34800074>.
>